### ISSUE TITLE
x-Day QuoteBarConsolidator reports incorrect Time and EndTime

### ISSUE THREAD

**Issue by Doggie52** [QuantConnect/Lean#3062]
#### Expected Behavior
For an _x_-day bar consolidated on midnight T, `EndTime` should be midnight T and `Time` should be T-_x_.

#### Actual Behavior
An _x_-day bar consolidated on midnight T has `EndTime` T-1 and `Time` T-1-_x_. The data itself seems correct (matches the data in .csv, assuming the leftmost column is open time) but I don't know what possible problems arise from having incorrect times.

#### Reproducing the Problem
```c#
namespace QuantConnect.Algorithm.CSharp
{
	public class BasicTemplateAlgorithm : QCAlgorithm
	{
		public QuoteBarConsolidator Consol;

		public override void Initialize()
		{
        		SetStartDate(2019,1,1);
        		SetEndDate(2019,2,1);

			var sec = AddForex("EURUSD", Resolution.Daily, Market.Oanda);
			Consol = new QuoteBarConsolidator(TimeSpan.FromDays(1));
			SubscriptionManager.AddConsolidator(sec.Symbol, Consol);

			Consol.DataConsolidated += (sender,data) => Log($"[{UtcTime.ToString(DateFormat.Forex)} UTC] EURUSD bar CONSOLIDATED with close time {data.EndTime.ToString(DateFormat.Forex)} and ask close {data.Ask.Close}.");
		}

		public override void OnData(Slice slice)
		{
			dynamic dynSlice;

			if ( slice.TryGetValue( "EURUSD", out dynSlice ) ) {
				Log($"[{UtcTime.ToString(DateFormat.Forex)} UTC] EURUSD bar RECEIVED with close time {dynSlice.EndTime.ToString(DateFormat.Forex)} and ask close {dynSlice.Ask.Close}.");
			}
		}
	}
}
```

The output will be something like the following:
```
[20190103 00:00:00.0000 UTC] EURUSD bar CONSOLIDATED with close time 20190102 00:00:00.0000 and ask close 1.1313.
[20190103 00:00:00.0000 UTC] EURUSD bar RECEIVED with close time 20190102 19:00:00.0000 and ask close 1.1313.
[20190104 00:00:00.0000 UTC] EURUSD bar CONSOLIDATED with close time 20190103 00:00:00.0000 and ask close 1.13908.
[20190104 00:00:00.0000 UTC] EURUSD bar RECEIVED with close time 20190103 19:00:00.0000 and ask close 1.13908.
[20190105 00:00:00.0000 UTC] EURUSD bar CONSOLIDATED with close time 20190104 00:00:00.0000 and ask close 1.13983.
[20190105 00:00:00.0000 UTC] EURUSD bar RECEIVED with close time 20190104 19:00:00.0000 and ask close 1.13983.
```

As you can see, the data _received_ has the correct close time, but the data _consolidated_ does not.

If you set the consolidator to be 2 days, output will be the following:
```
[20190115 00:00:00.0000 UTC] EURUSD bar RECEIVED with close time 20190114 19:00:00.0000 and ask close 1.14736.
[20190116 00:00:00.0000 UTC] EURUSD bar RECEIVED with close time 20190115 19:00:00.0000 and ask close 1.14147.
[20190117 00:00:00.0000 UTC] EURUSD bar CONSOLIDATED with close time 20190116 00:00:00.0000 and ask close 1.14004.
[20190117 00:00:00.0000 UTC] EURUSD bar RECEIVED with close time 20190116 19:00:00.0000 and ask close 1.14004.
```

#### System Information
QC Web Terminal

#### Related Issues
Possibly #1569 but unsure which issue that one referred to when it was closed.

#### Checklist

- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

- [x] I have provided detailed steps to reproduce the issue

**Comment by Doggie52** [comment]
I think what's happening here is the following:

At midnight T UTC: new `_workingBar` is created.

Oanda FX exchange TZ is `New_York` so `data.Time` is given as T-4h (or 5h)

Thus `Time = GetRoundedBarTime(data.Time)` will end up being rounded down to midnight T-1.

At midnight T+1 when bar gets consolidated, `EndTime` is just `Time + Period` which will then end up being midnight T.

**Comment by Martin-Molinero** [comment]
Hey!

> If you set the consolidator to be 2 days, output will be the following:

This case was solved some time ago by https://github.com/QuantConnect/Lean/pull/4940 no more rounding it happening.

> var sec = AddForex("EURUSD", Resolution.Daily, Market.Oanda);
 Consol = new QuoteBarConsolidator(TimeSpan.FromDays(1));
 
 This other case is trickier, Lean still rounds down TimeSpans of a day or smaller. I've pushed some adjustments at https://github.com/QuantConnect/Lean/pull/6408 for these cases
